iaso/api/entity.py

```diff
@@ -142,32 +149,42 @@ def export_entity_as_csv(entities):
     header = []
     data = []
     filename = ""
-
+    possible_field_list = []
+    # NOT WORKING ATM
     for entity in entities:
-        res = {"entity": EntitySerializer(entity, many=False).data}
-        benef_data = []
-        for k, v in res["entity"].items():
-            try:
-                fields_list = entity.entity_type.fields_list_view
-                for f in fields_list:
-                    if f is None:
-                        raise serializers.ValidationError(
-                            {"error": "You must provide a field view list in order to export the entities."}
-                        )
-            except TypeError:
-                raise serializers.ValidationError(
-                    {"error": "You must provide a field view list in order to export the entities."}
-                )
-            if k in fields_list or k == "attributes":
-                if k == "attributes":
-                    for k_, v_ in res["entity"]["attributes"]["file_content"].items():
-                        if k_ not in header:
-                            header.append(k_)
-                        benef_data.append(v_)
+        res = entity.attributes.json
+        benef_data = [None]
+        if res is not None:
+            for k, v in res.items():
+                try:
+                    fields_list = entity.entity_type.fields_list_view
+                    for f in fields_list:
+                        if f is None:
+                            raise serializers.ValidationError(
+                                {"error": "You must provide a field view list in order to export the entities."}
+                            )
+                except TypeError:
+                    raise serializers.ValidationError(
+                        {"error": "You must provide a field view list in order to export the entities."}
+                    )
+
+                for f in entity.attributes.form.possible_fields:
+                    for k_, v_ in f.items():
+                        if k_ == "name" and v_ not in possible_field_list:
+                            possible_field_list.append(v_)
+                for h in possible_field_list:
+                    if h in fields_list and h not in header:
+                        header.append(h)
+        i = 0
+        for h in header:
+            for k, v in res.items():
+                match = False
+                if k == h:
+                    benef_data.append(f"found at {k}")
+                if i < len(header):
+                    i += 1
```

Comment:
The csv doesnt work in this version as I have to finish it. So this part of the code is a WIP.